packages/truffle-debugger/lib/evm/reducers.js

```diff
@@ -14,27 +14,55 @@ function contexts(state = DEFAULT_CONTEXTS, action) {
      * Adding a new context
      */
     case actions.ADD_CONTEXT:
-      let { contractName, binary } = action;
+      const addContext = ({ contractName, raw }) => {
```

Comment:
The style here is confusing. For both the ADD_CONTEXT case and the ADD_BINARY case, you define a function in the body of the case and then immediately use it, just once. Why define a function at all, if it's never used again? If it's going to be used again, why define it just here? I think this would be clearer just written out without the extra functions.

Reply:
I was just trying this out. We have a couple switch statements that declare a bunch of variables, and I'm sick of having to move the variable declarations up outside the switch statement. So here I'm trying out the pattern for a particular case:

```javascript
case something:
  const inlineFunc = ({ /* action... */ }) => {
    /* ... */
  }
  
  return inlineFunc(action)
```

Not sure. What do you think?